quotes/static/quotes/quotes_list.js

```diff
@@ -12,11 +11,23 @@ document.addEventListener('DOMContentLoaded', function(){
                 .then(function(response){
                     // `element` is the vote up button, and its sibling with class .quote__sum contains
                     // the actual quote sum data.
-                    element.parentNode.querySelector('.quote__sum').innerHTML = response.data.sum;
+                    element.parentNode.querySelector('.quote__sum').innerHTML = "Score: " + response.data.sum;
                 })
 
         });
     });
+    var approveQuote = document.querySelectorAll(".quote__approve");
+    approveQuote.forEach(function(element){
+        var quoteId = element.getAttribute('data-quote-id');
+        element.addEventListener('click', function(){
+            axios('/internal/quotes/' + quoteId + '/approve', {method: 'POST'})
+                .then(function(response){
+                    // `element` is the vote down button, and its sibling with class .quote__sum contains
+                    // the actual quote sum data.
+                    location.reload();
```

Comment:
In my opinion: No.

The reason I wrote the upvote/downvote functionality using AJAX, is that I found upvoting or downvoting to be sufficiently small operations to not warrant a full page reload. Also, only a single piece of new information needs to be re-fetched: The new total sum. Note that the views for voting up and voting down return `JsonResponse` objects with the updated sum.

It is very easy to update the rendered content to reflect this, as can be seen in the inner functions:

```
element.parentNode.querySelector('.quote__sum').innerHTML = "Score: " + response.data.sum;
```

For approving, I feel AJAX is the wrong strategy to go. The fact that we actually call `location.reload` makes this very clear. AJAX is useful in exactly the instances where we do not want to reload the full page. That we have to call `location.reload` is a very strong indicator that we should not have used AJAX in the first place. Some more points were we to actually use AJAX properly for approving:

* On successful approval we will have to remove the div from the list unapproved objects and insert it manually into the list of approved objects. This is neither hard nor slow but does require some JS twiddling.
* Using AJAX might give a false sense of reactiveness: We will not get information regarding new approved/unapproved quotes from the server since the last page load, other than for the quote we have now approved. In the meanwhile, other quotes might have been approved by others, some might have been deleted, and so forth. This is automatically solved when doing a full page reload.
* Approving something will eventually require some sort of permission. The endpoint should handle cases where the user does not have sufficient permissions, by redirecting to some page warning of this, or use the [messages framework](https://docs.djangoproject.com/en/2.1/ref/contrib/messages/). This would instead have to be handled explicitly in our axios callback if we were to use AJAX.

Note that none of the above cases applies (again, in my opinion) for the upvote/downvote question. Apart perhaps from the last part of the last point; we will have to do some manual error handling for the upvote downvote calls at some point.